src/app/SOCestimator/SOCestimator.c

```diff
@@ -0,0 +1,134 @@
+#include <stdbool.h>
+
+#include "common_macros.h"
+#include "BatteryCharacteristics.h"
+#include "SOCestimator.h"
+
+#define CELL_CAPACITY_Ah 6.550
+#define BATTERY_CAPACITY_Ah (NUM_SERIES_CELLS * CELL_CAPACITY_Ah)
+
+int SOC_raw;
+int SOC_limit;
+
+/**
+ * Use the average series cell voltage and ambient temperature to determine a maximum SOC.
+ * Voltage thresholds determined (shittily) by eyeballing voltage/discharge % graph on the cell datasheet.
+ * avg_V [in] - average series cell voltage
+ * ambient_temp_C [in] - ambient temperature
+ */
+static int SOC_limit_from_voltage(float avg_V, float ambient_temp_C)
+{
+    // determine which temperature range the ambient temp falls into
+    // saturate to safe values
+    int temp_index = SAT((ambient_temp_C + .5) / 10, 0, NUM_TEMP_RANGES - 1);
+    
+    if (avg_V < SOC_25_LIMIT_FOR_TEMP_INDEX(temp_index))
```

Comment:
Similar to a previous comment but why do we choose 25, 50, and 75? Especially since they are hardcoded in